packages/tendermint-rpc/src/dates.ts

```diff
@@ -0,0 +1,39 @@
+import { fromRfc3339 } from "@cosmjs/encoding";
+import { ReadonlyDate } from "readonly-date";
+
+export interface ReadonlyDateWithNanoseconds extends ReadonlyDate {
+  /* Nanoseconds after the time stored in a vanilla ReadonlyDate (millisecond granularity) */
+  readonly nanoseconds?: number;
+}
+
+export interface DateWithNanoseconds extends Date {
+  /** Nanoseconds after the time stored in a vanilla Date (millisecond granularity) */
+  nanoseconds?: number;
+}
+
+export function fromRfc3339WithNanoseconds(dateTimeString: string): DateWithNanoseconds {
```

Comment:
Nice to return the more permissive variant, which can "down cast" to the Readonly variant